src/station-race.ts

```diff
@@ -196,13 +196,13 @@ const turn = (state: Setup): Turn => {
 };
 
 const getOffTheTrain = (state: Turn): GameOver | TurnResult =>
-  hasWinner(state) ? gameOver(state) : turnResult(state);
+  isCurrentPlayerWinner(state) ? gameOver(state) : turnResult(state);
```

Comment:
This is pretty cool! And makes sense since you only have a winner when the game is over (i.e. `GameOver` state). It also removes the nullable return type for `winner`, which was a smell to begin with 👍 pretty cool!